tests/rust-apps-wasm/add-comp/src/lib.rs

```diff
@@ -1,6 +1,14 @@
 #![no_std]
 
-cargo_component_bindings::generate!();
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+#[panic_handler]
+fn my_panic(_info: &core::panic::PanicInfo) -> ! {
+    loop {}
```

Comment:
I think you end up with even better codegen if you use `core::intrinsics::abort()` here, though that comes with the need for `#![feature(core_intrinsics)]`